internal/pkg/agent/application/actions/handlers/handler_action_upgrade.go

```diff
@@ -31,12 +34,34 @@ func NewUpgrade(log *logger.Logger, coord *coordinator.Coordinator) *Upgrade {
 }
 
 // Handle handles UPGRADE action.
-func (h *Upgrade) Handle(ctx context.Context, a fleetapi.Action, _ acker.Acker) error {
+func (h *Upgrade) Handle(ctx context.Context, a fleetapi.Action, ack acker.Acker) error {
 	h.log.Debugf("handlerUpgrade: action '%+v' received", a)
 	action, ok := a.(*fleetapi.ActionUpgrade)
 	if !ok {
 		return fmt.Errorf("invalid type, expected ActionUpgrade and received %T", a)
 	}
-
-	return h.coord.Upgrade(ctx, action.Version, action.SourceURI, action)
+	go func() {
```

Comment:
Do we want to have sync/async handlers, or should we change how the gateway sends actions to the dispatcher?
The diagnostics action that is being added may also need to run async in case of a poor network connection

Reply:
I don't have a strong opinion either way.

Reply:
I do not believe we should change how the gateway sends actions, because some need to be synchronous but others might need to be async (like this one, and probably diagnostics).

I am okay with this being internal to the action, but a uniform way of defining a sync action and an async action would be nice.